src/databricks/sql/cloudfetch/download_manager.py

```diff
@@ -97,9 +98,11 @@ def get_next_downloaded_file(
             self.download_handlers.pop(idx)
             # Return True upon successful download to continue loop and not force a retry
             return result
-        # Download was not successful for next download item, force a retry
+        # Download was not successful for next download item. Fail
```

Comment:
The original code retries immediately. What is the difference?

Reply:
the original code was wrong. Now I am properly retrying.

Reply:
and most importantly, actually failing. We cannot continue if a file fails to download after the retries

Reply:
The original code returns None in case of failure?